submissions/usides/friends-app/js/main.js

```diff
@@ -0,0 +1,149 @@
+  filterHumans(currentState.filter);
+  sortHumans(currentState.sort);
+  updateCards(currentState.humans);
+};
+
+const handleOptionClick = function (e) {
+  const button = e.target.closest('button');
+  if (!button) return;
+  if (button.dataset.sort) {
+    sortButtons.forEach((button) => button.classList.remove('active'));
+    button.classList.add('active');
+    sortHumans(button.dataset.sort);
+  }
+  if (button.dataset.filter) {
+    if (currentState.filter === button.dataset.filter) {
+      button.classList.toggle('active');
+      filterHumans('none');
+    } else {
+      filterButtons.forEach((button) => button.classList.remove('active'));
+      button.classList.add('active');
+      filterHumans(button.dataset.filter);
+    }
+    sortHumans(currentState.sort);
+  }
+  updateCards(currentState.humans);
+};
+
+const filterHumans = function (filterOption) {
+  const searchedName = humansOriginState.filter((elem) =>
+    elem.name.first.toLowerCase().startsWith(currentState.searchValue),
+  );
+  if (filterOption === 'none') {
+    currentState.humans = searchedName;
+  } else {
+    currentState.humans = searchedName.filter(
+      (elem) => elem.gender === filterOption,
+    );
+  }
+  currentState.filter = filterOption;
+};
+
+const sortByAge = (a, b) => (a.dob.age <= b.dob.age ? 1 : -1);
+const sortByName = (a, b) =>
+  a.name.first.toLowerCase() >= b.name.first.toLowerCase() ? 1 : -1;
+
+const sortHumans = function (sortOption) {
+  switch (sortOption) {
+    case 'a-z':
+      currentState.humans.sort((a, b) => sortByName(a, b));
+      break;
+
+    case 'z-a':
+      currentState.humans.sort((a, b) => sortByName(b, a));
+      break;
+
+    case 'young':
+      currentState.humans.sort((a, b) => sortByAge(b, a));
+      break;
+
+    case 'old':
+      currentState.humans.sort((a, b) => sortByAge(a, b));
+      break;
+  }
+
+  currentState.sort = sortOption;
+};
+
+const updateCards = function (arr) {
+  cardsHolder.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+  arr.forEach((elem) => {
+    const newElem = document.createElement('div');
+    newElem.classList.add('card');
+    const template = `
+    <p class="u-name">${elem.name.first} ${elem.name.last}</p>
+    <div class="splitter">
+      <img src="${elem.picture.large}" alt="User Photo">
+      <div class="data-group">
+        <p class="label">Age</p>
+        <p class="data">${elem.dob.age}</p>
+        <p class="label">Gender</p>
+        <p class="data">${elem.gender}</p>
+      </div>
+    </div>`;
+    newElem.innerHTML = template;
+    fragment.appendChild(newElem);
+  });
+  cardsHolder.appendChild(fragment);
+};
+
+const getHumans = function () {
+  return fetch(HUMANS_LINK)
+    .then((res) => {
+      if (res.status !== 200) {
+        throw new Error(`Couldn't fetch data. ${res.status}`);
+      }
+      return res.json();
+    })
+    .then((data) => {
+      humansOriginState = data.results;
```

Comment:
do you really need to create this variable?

Reply:
Yeah. I need it. I need untouched human array each time when I remove a letter from input field (I cant use currentState as it has filtered array).